server/passport-setup.js

```diff
@@ -0,0 +1,27 @@
+const passport = require('passport');
+const GoogleStrategy = require('passport-google-oauth').OAuth2Strategy;
+const keys = require('./keys');
+const app = require("./server");
+const User = app.models.User; 
+
+passport.serializeUser((user, done) => {
+  done(null, user.id);
+});
+
+passport.deserializeUser((id, done) => {
+  User.findById(id).then((user) => {
+      done(null, user);
+  });
+});
+
+passport.use(new GoogleStrategy({
+    clientID: keys.google.clientID,
+    clientSecret: keys.google.clientSecret,
+    callbackURL: "/auth/google/redirect"
+  },
+  function(accessToken, refreshToken, profile, done) {
+       User.findOrCreate({ googleId: profile.id, username: profile.displayName, email: profile._json.email }, function (err, user) {
+         return done(err, user);
```

Comment:
this is definitely a nit: you can pass done as the callback : 
` User.findOrCreate({ googleId: profile.id, username: profile.displayName, email: profile._json.email }, done)